frontend/lib/resolution/resolution-queries.cpp

```diff
@@ -4049,6 +4047,27 @@ static bool resolveFnCallSpecial(Context* context,
                                     BoolParam::get(context, result));
         return true;
       }
+    } else if (ci.numActuals() == 1 && ci.hasQuestionArg()) {
+      // support type and param comparisons with '?'
+      // TODO: will likely need adjustment once we are able to compare a
+      // partially-instantiated type's fields with '?'
+      auto arg = ci.actual(0).type();
+      bool result = false;
+      bool haveResult = true;
+      if (arg.isType()) {
+        result = arg.type()->isAnyType();
+      } else if (arg.isParam()) {
+        result = arg.param() == nullptr;
+      } else {
+        haveResult = false;
+      }
+      result = ci.name() == USTR("==") ? result : !result;
+      if (haveResult) {
+        exprTypeOut =
+            QualifiedType(QualifiedType::PARAM, BoolType::get(context),
+                          BoolParam::get(context, result));
+        return true;
+      }
```

Comment:
I could've sworn we had a helper for creating QTs from bools and other values. I couldn't find it, though.

Reply:
I don't think we do. There's definitely a lot of examples of writing it out like this, for example in prims.cpp. If we did have a helper, where would it go? On `QualifiedType`?

Reply:
> Not saying you should do anything as part of this PR

Yet I shall